cpp/src/join/joining.cu

```diff
@@ -253,260 +220,345 @@ gdf_error join_call( int num_cols, gdf_column **leftcol, gdf_column **rightcol,
+    
+    if (right_nonjoin_indices.size() > 0)
+        std::set_difference(std::cbegin(right_column_indices), std::cend(right_column_indices),
+                                 std::cbegin(tmp_right_join_indices), std::cend(tmp_right_join_indices),
+                                 std::begin(right_nonjoin_indices));
 
     gdf_size_type join_size = left_indices->size;
-    int left_table_end = num_left_cols - num_cols_to_join;
-    int right_table_begin = num_left_cols;
-
-    //create left and right output column data buffers
-    for (int i = 0; i < left_table_end; ++i) {
-        gdf_column* col = lnonjoincol[i];
-        gdf_column_view_augmented(result_cols[i], nullptr, nullptr, join_size,
-                                  col->dtype, 0, col->dtype_info, col->col_name);
-        int col_width = cudf::byte_width(*(result_cols[i]));
-        RMM_TRY( RMM_ALLOC((void**)&(result_cols[i]->data), col_width * join_size, 0) ); // TODO: non-default stream?
-        RMM_TRY( RMM_ALLOC((void**)&(result_cols[i]->valid), sizeof(gdf_valid_type)*gdf_valid_allocation_size(join_size), 0) );
-        CUDA_TRY( cudaMemset(result_cols[i]->valid, 0, sizeof(gdf_valid_type)*gdf_valid_allocation_size(join_size)) );
-        CHECK_STREAM(0);
-    }
-    for (int i = right_table_begin; i < result_num_cols; ++i) {
-        gdf_column* col = rnonjoincol[i - right_table_begin];
-        gdf_column_view_augmented(result_cols[i], nullptr, nullptr, join_size,
-                                  col->dtype, 0, col->dtype_info, col->col_name);
-        int col_width = cudf::byte_width(*(result_cols[i]));
-        RMM_TRY( RMM_ALLOC((void**)&(result_cols[i]->data), col_width * join_size, 0) ); // TODO: non-default stream?
-        RMM_TRY( RMM_ALLOC((void**)&(result_cols[i]->valid), sizeof(gdf_valid_type)*gdf_valid_allocation_size(join_size), 0) );
-        CUDA_TRY( cudaMemset(result_cols[i]->valid, 0, sizeof(gdf_valid_type)*gdf_valid_allocation_size(join_size)) );
-        CHECK_STREAM(0);
-    }
-    //create joined output column data buffers
-    for (int join_index = 0; join_index < num_cols_to_join; ++join_index) {
-        int i = left_table_end + join_index;
-        gdf_column* col = left_cols[left_join_cols[join_index]];
-        gdf_column_view_augmented(result_cols[i], nullptr, nullptr, join_size,
-                                  col->dtype, 0, col->dtype_info, col->col_name);
-        int col_width = cudf::byte_width(*(result_cols[i]));
-        RMM_TRY( RMM_ALLOC((void**)&(result_cols[i]->data), col_width * join_size, 0) ); // TODO: non-default stream?
-        RMM_TRY( RMM_ALLOC((void**)&(result_cols[i]->valid), sizeof(gdf_valid_type)*gdf_valid_allocation_size(join_size), 0) );
-        CUDA_TRY( cudaMemset(result_cols[i]->valid, 0, sizeof(gdf_valid_type)*gdf_valid_allocation_size(join_size)) );
-        CHECK_STREAM(0);
+    // Update first set of type and type infos from left
+    std::vector <gdf_dtype> rdtypes = cudf::column_dtypes(left);
```

Comment:
Ambiguous abbreviation. Don't put spaces between a type and it's template arguments. 
```suggestion
    std::vector<gdf_dtype> result_dtypes = cudf::column_dtypes(left);
```